File: src/IsbInterop/Data/IPickRequisiteItems.vb
Imports IsbInterop.Accessory

Namespace Data
  ''' <summary>
  ''' Список описаний допустимых значений реквизита типа «Признак».
  ''' </summary>
  Public Interface IPickRequisiteItems
    Inherits IForEach(Of IPickRequisiteItem)

    ''' <summary>
    ''' Получить ИД по значению.
    ''' </summary>
    ''' <param name="namedValue">Допустимое значение реквизита.</param>
    ''' <returns>ИД допустимого значения реквизита типа «Признак» – 
    ''' внутреннее значение реквизита, хранящееся на сервере.</returns>
    Function IdByValue(namedValue As String) As String
  End Interface
End NameSpace

File: src/IsbInterop/Data/IPickRequisiteItems.cs
using IsbInterop.Accessory;

namespace IsbInterop.Data
{
  /// <summary>
  /// Список описаний допустимых значений реквизита типа «Признак».
  /// </summary>
  public interface IPickRequisiteItems : IForEach<IPickRequisiteItem>
  {
    /// <summary>
    /// Получить ИД по значению.
    /// </summary>
    /// <param name="namedValue">Допустимое значение реквизита.</param>
    /// <returns>ИД допустимого значения реквизита типа «Признак» –
    /// внутреннее значение реквизита, хранящееся на сервере.</returns>
    string IdByValue(string namedValue);
  }
}
